src/main/java/smartthings/ratpack/sqs/DefaultSqsService.java

```
@@ -35,7 +37,8 @@ public DefaultSqsService(
     @Override
     public void onStart(StartEvent event) throws Exception {
         if (config.isEnabled()) {
-            this.sqs = sqsConfigService.getAmazonSQSAsync(config.getRegion());
+            sqs = sqsConfigService.getAmazonSQSAsync(config.getRegion());
+            Optional.ofNullable(config.getEndpoint()).ifPresent(sqs::setEndpoint);
```

What will happen if the service is enabled but the endpoint is not provided?  If this will just fail later on it might be better to throw some kind of validation exception rather than ignoring the missing parameter.

---

Another possibility would be to provide a means of overriding all these default implementations.  Otherwise, we have to potentially maintain all the config options AWS's Sqs client may support.

I was thinking we could do something like this in the SqsModule
```
    @Override
    protected void configure() {
        OptionalBinder.newOptionalBinder(binder(), CircuitBreaker.class)
            .setDefault().toProvider(SimpleCircuitBreaker::new);

        OptionalBinder.newOptionalBinder(binder(), SqsService.class)
            .setDefault().to(DefaultSqsService.class).in(Scopes.SINGLETON);

        OptionalBinder.newOptionalBinder(binder(), SqsConsumerService.class)
            .setDefault().to(DefaultSqsConsumerService.class).in(Scopes.SINGLETON);

        OptionalBinder.newOptionalBinder(binder(), SqsConsumerService.class)
            .setDefault().to(DefaultSqsConsumerService.class).in(Scopes.SINGLETON);

        OptionalBinder.newOptionalBinder(binder(), SqsConfigService.class)
            .setDefault().to(DefaultSqsConfigService.class).in(Scopes.SINGLETON);
    }
```

That way anyone could just provide an alternate implementation of SqsConfigService which would allow user to build any SQS client they want to be used.

---

Also, @timic this library never really got out of the prototype stage.  So there are likely other issues.  I started it, but my use-case disappeared.

---

@llinder 
Endpoint is optional. In case it isn't provided client will  automatically select amazon server relative to configured aws credentials and region.

---

@jeff-blaisdell 
I researched the existing libraries for integrating ratpack with sqs. I found that this one requires configured aws sqs service which can be unhandy in development stage and in writing integration tests. So this option should be configured as simple as possible. I agree that optional binder is good approach but this is a topic to another pull request.

I choosed spring boot with aws cloud messaging autoconfigurer. But I hope to switch it to this one when it will become ready to production.

---

@jeff-blaisdell I agree in that there are way to many config options to cover all use cases with config parameters.  Setting up default config and allowing apps to override instantiation of the underlying client seems like an ideal approach.  The optional endpoint parameter that @timic proposed here seems ok.  I originally thought this was the actual queue URL but after re-reading AWS docs the use case for setting this is a clearer and seems like a useful config value to include.

---

I thought same on endpoint.  Yeah, on second thought that does seem like a sensible one to provide out-of-box.  This looks good to me as well.